### PR TITLE
fix(agents): keep internal markers and raw executor text out of user replies

### PR DESCRIPTION
**Pull request by sankalpaacharya**
## Summary
Live on iMessage (any bot platform, actually) a user saw two internal artifacts in one turn: a reply starting with the literal `[EXECUTOR_RESULT]` block, and a separate bubble with the executor's raw terminal text ("The user does not have a WhatsApp integration connected…"). Root causes, both fixed here:

- **Comms streamed a parroted `[EXECUTOR_RESULT]` block** — a weak model imitated the internal format its prompt describes. `strip_internal_agent_markers` only ran on narrated text, never on the streamed comms reply. Now every streamed comms chunk goes through a stream-safe `InternalMarkerFilter` (markers removed even when split across chunks; only a possible marker prefix is ever held back; flushed on stream end and on cancellation). The batch strip is now the same implementation.
- **Narration fell back to raw executor text** — `narrate_executor_result` runs the comms graph, which selects its model by `configurable["provider"]` and has no provider failover; when OpenRouter returned 402 the narration died and `_narrate_and_deliver` shipped the raw text. Narration now catches `LLM_FALLBACK_EXCEPTIONS`, re-pins to the next configured provider (`next_fallback_provider` / `pin_fallback_provider`, same priority order the direct-call helpers use) and retries once; the raw-text fallback remains only if every provider fails.

Also fixed while getting the mutation lane green: `scripts/test/mutation_classify.py` crashed on class-method mutants (mutmut writes the orig name qualified as `Class.xǁClassǁmethod__mutmut_orig`) and, once past that, would have judged every method mutant EQUIV because indented method bodies read as empty. It now parses dotted names, extracts indented bodies, and fails closed on an empty body.

## Tests
Red-first regression tests: `test_parroted_marker_split_across_chunks_is_stripped_from_the_stream`, `test_provider_failure_retries_the_narration_on_the_fallback_provider` (both proven red on master by `regression-proof`); filter edge cases (chunk splits, every prefix length, whitespace, cancellation flush); provider fallback selection/pinning; classifier subprocess tests (red on master's script). Mutation lane: no survivors on changed lines in `comms_narrator.py`, `plan_model.py`, `client.py`, `agent_helpers.py`, `agent_utils.py`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01NJsJ8rdW1qyNFdiZoTJkTL

### CODE REVIEW

File: scripts/test/mutation_classify.py [hunk 1]
```diff
@@ -27,13 +27,17 @@
...
-orig_match = re.search(rf"^{re.escape(dict_name)}\['_mutmut_orig'\]\s*=\s*(\w+)", src, re.MULTILINE)
+orig_match = re.search(rf"^{re.escape(dict_name)}\['_mutmut_orig'\]\s*=\s*([\w.]+)", src, re.MULTILINE)
 if not orig_match:
     sys.exit(1)
-orig_name = orig_match.group(1)
-blocks = re.split(r"^(?:async )?def ", src, flags=re.MULTILINE)
+orig_name = orig_match.group(1).rsplit(".", 1)[-1]
+# Methods are indented, so the split must tolerate leading whitespace — otherwise
+# a method's body reads as empty and orig == mutant, a false EQUIV verdict.
+blocks = re.split(r"^[ \t]*(?:async )?def ", src, flags=re.MULTILINE)
```
[the remark this answers is not in the document]
Comment (sankalpaacharya) [reply scripts/test/mutation_classify.py]: Fixed in 3a4246013.

The def-split was the wrong tool — replaced with an AST lookup of the uniquely-named mutmut function, so a mutant body is now `src_lines[node.lineno:node.end_lineno]` and a nested `def` is simply part of that body rather than a split point. Indented methods, nested functions, and blank lines are all handled by construction now.

Red-first regression test added: `TestMethodMutants::test_a_change_after_a_nested_function_is_still_seen` in `apps/api/tests/unit/scripts/test_mutation_classify.py` — a method containing a nested `def` with the mutation *after* it. Confirmed failing on the previous regex implementation and passing on the fix. Also re-ran the mutation lane against `app/utils/agent_utils.py` (the class-method-heavy module in this PR): no survivors on changed lines.

**Comment by aryanranderiya** [comment]
Closing as duplicate of #1020 (fixed)
